src/ScheduledTask/FetchPaymentMethodLogosHandler.php

```diff
@@ -0,0 +1,180 @@
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2020 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\ScheduledTask;
+
+use Adyen\Shopware\PaymentMethods\PaymentMethodInterface;
+use Adyen\Shopware\PaymentMethods\PaymentMethods;
+use Adyen\Shopware\Service\ConfigurationService;
+use Exception;
+use Psr\Log\LoggerAwareTrait;
+use Shopware\Core\Checkout\Payment\PaymentMethodEntity;
+use Shopware\Core\Content\Media\File\MediaFile;
+use Shopware\Core\Content\Media\MediaService;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Filter\EqualsFilter;
+use Shopware\Core\Framework\MessageQueue\ScheduledTask\ScheduledTaskHandler;
+use Symfony\Component\HttpFoundation\Request;
+
+class FetchPaymentMethodLogosHandler extends ScheduledTaskHandler
+{
+    use LoggerAwareTrait;
+
+    /**
+     * @var ConfigurationService
+     */
+    private $configurationService;
+    /**
+     * @var MediaService
+     */
+    private $mediaService;
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $paymentMethodRepository;
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $mediaRepository;
+    /**
+     * @var bool
+     */
+    private $enableUrlUploadFeature;
+
+    public function __construct(
+        EntityRepositoryInterface $scheduledTaskRepository,
+        ConfigurationService $configurationService,
+        MediaService $mediaService,
+        EntityRepositoryInterface $paymentMethodRepository,
+        EntityRepositoryInterface $mediaRepository,
+        bool $enableUrlUploadFeature = true
+    ) {
+        parent::__construct($scheduledTaskRepository);
+        $this->configurationService = $configurationService;
+        $this->mediaService = $mediaService;
+        $this->paymentMethodRepository = $paymentMethodRepository;
+        $this->mediaRepository = $mediaRepository;
+        $this->enableUrlUploadFeature = $enableUrlUploadFeature;
+    }
+
+    public static function getHandledMessages(): iterable
+    {
+        return [ FetchPaymentMethodLogos::class ];
+    }
+
+    public function run(): void
+    {
+        if (!$this->enableUrlUploadFeature) {
+            $this->logger->debug('Configuration `shopware.media.enable_url_upload_feature` is disabled.');
```

Comment:
good one +1
Is it a config that the merchant needs to enable to make sure they can use this?
Please add this as a step to the release note and as a suggestion to docs (let me know if you need help with that ;) )

Reply:
Usually it is enabled by default but yes they do need to make sure it it enabled if they want logos